src/libsyntax/util/small_vector.rs

```diff
@@ -69,6 +69,42 @@ impl<T> SmallVector<T> {
         }
     }
 
+    pub fn pop(&mut self) -> Option<T> {
+        match self.repr {
+            Zero => None,
+            One(..) => {
+                let one = mem::replace(&mut self.repr, Zero);
+                match one {
+                    One(v1) => Some(v1),
+                    _ => unreachable!()
+                }
+            }
+            Many(..) => {
+                let mut many = mem::replace(&mut self.repr, Zero);
+                let item =
+                    match many {
```

Comment:
We don't necessarily need to worry too much about making sure to convert a length 1 vec to a `One`. SmallVector is trying to avoid the initial allocation, but if that's already been made, keeping the vec around isn't that big of a deal.